pkg/operator/secretannotator/status/status.go

```diff
@@ -16,36 +16,46 @@ import (
 	"github.com/openshift/cloud-credential-operator/pkg/util/clusteroperator"
 )
 
+type SecretStatusHandler struct {
+	client.Client
+}
+
+func NewSecretStatusHandler(kubeClient client.Client) *SecretStatusHandler {
+	return &SecretStatusHandler{
+		Client: kubeClient,
+	}
+}
+
+var _ clusteroperator.StatusHandler = &SecretStatusHandler{}
+
 // SyncOperatorStatus computes the operator's current status and
 // creates or updates the ClusterOperator resource for the operator accordingly.
-func SyncOperatorStatus(kubeClient client.Client) error {
+func (s *SecretStatusHandler) SyncOperatorStatus() error {
```

Comment:
So that the reconcilers do not have to carry around the `StatusHandler` object.
```suggestion
func SyncOperatorStatus(c client.Client) error {
```